src/Npgsql/NpgsqlReadBuffer.Stream.cs

```diff
@@ -158,12 +158,50 @@ async Task<int> ReadLong(ValueTask<int> task, bool async)
                 return read;
             }
 
+#if !NET461 && !NETSTANDARD2_0
+            public override int Read(Span<byte> span)
+                => Read(span.ToArray(), CancellationToken.None, false).GetAwaiter().GetResult();
+
+            public override ValueTask<int> ReadAsync(Memory<byte> buffer, CancellationToken cancellationToken)
+                => Read(buffer, cancellationToken, true);
+
+            ValueTask<int> Read(Memory<byte> buffer, CancellationToken cancellationToken, bool async)
+            {
+                CheckDisposed();
+
+                if (cancellationToken.IsCancellationRequested)
+                    return new ValueTask<int>(Task.FromCanceled<int>(cancellationToken));
+
+                var count = Math.Min(buffer.Length, _len - _read);
+
+                if (count == 0)
+                    return new ValueTask<int>(0);
+
+                var task = _buf.ReadBytes(buffer.Slice(0, count), async);
+                if (task.IsCompleted) // This may be a bug in the new version of ValueTask
+                {
+                    _read += task.Result;
+                    return task;
+                }
+
+                return ReadLong(task, async);
+            }
+#endif
+
             public override void Write(byte[] buffer, int offset, int count)
                 => throw new NotSupportedException();
 
             public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
                 => throw new NotSupportedException();
 
+#if !NET461 && !NETSTANDARD2_0
+            public override void Write(ReadOnlySpan<byte> span)
+                => throw new NotSupportedException();
```

Comment:
Unless I'm mistaken, this isn't necessary - doesn't the default implementation in Stream call the byte[] overload, which would throw the exception anyway?

Reply:
Yes it does but after allocating memory and copying user data into it...
I can remove it, but then I also remove the WriteAsync(byte[]) 😋

Reply:
I don't think an allocation and copy matters much if the method ends up throwing NotSupportedException... Makes sense to me to remove WriteAsync too.

Reply:
Removed.